plugins/range/plugin.go

```diff
@@ -0,0 +1,220 @@
+		if len(tokens) != 3 {
+			return nil, fmt.Errorf("malformed line, want 3 fields, got %d: %s", len(tokens), line)
+		}
+		hwaddr, err := net.ParseMAC(tokens[0])
+		if err != nil {
+			return nil, fmt.Errorf("malformed hardware address: %s", tokens[0])
+		}
+		ipaddr := net.ParseIP(tokens[1])
+		if ipaddr.To4() == nil {
+			return nil, fmt.Errorf("expected an IPv4 address, got: %v", ipaddr)
+		}
+		expires, err := time.Parse(time.RFC3339, tokens[2])
+		if err != nil {
+			return nil, fmt.Errorf("expected time of exipry in RFC3339 format, got: %v", tokens[2])
+		}
+		records[hwaddr.String()] = &Record{IP: ipaddr, expires: expires}
+	}
+	return records, nil
+}
+
+// Handler6 handles DHCPv6 packets for the file plugin
+func Handler6(req, resp dhcpv6.DHCPv6) (dhcpv6.DHCPv6, bool) {
+	// TODO add IPv6 netmask to the response
+	return resp, false
+}
+
+// Handler4 handles DHCPv4 packets for the range plugin
+func Handler4(req, resp *dhcpv4.DHCPv4) (*dhcpv4.DHCPv4, bool) {
+	record, ok := DHCPv4Records[req.ClientHWAddr.String()]
+	if !ok {
+		log.Printf("plugins/file: MAC address %s is new, leasing new IP address", req.ClientHWAddr.String())
+		rec, err := createIP(ipRangeStart, ipRangeEnd)
+		if err != nil {
+			log.Error(err)
+			return nil, true
+		}
+		err = saveIPAddress(req.ClientHWAddr, rec)
+		if err != nil {
+			log.Printf("plugins/file: SaveIPAddress failed: %v", err)
+		}
+		DHCPv4Records[req.ClientHWAddr.String()] = rec
+		record = rec
+	}
+	resp.YourIPAddr = record.IP
+	resp.Options.Update(dhcpv4.OptIPAddressLeaseTime(LeaseTime))
+	log.Printf("plugins/file: found IP address %s for MAC %s", record.IP, req.ClientHWAddr.String())
+	return resp, false
+}
+
+func setupRange6(args ...string) (handler.Handler6, error) {
+	// TODO setup function for IPv6
+	log.Warning("plugins/range: not implemented for IPv6")
+	return Handler6, nil
+}
+
+func setupRange4(args ...string) (handler.Handler4, error) {
+	_, h4, err := setupRange(false, args...)
+	return h4, err
+}
+
+func setupRange(v6 bool, args ...string) (handler.Handler6, handler.Handler4, error) {
+	if v6 {
+
+	} else {
+		if len(args) < 4 {
+			return nil, nil, errors.New("need a file name, server IP, netmask and a lease time")
+		}
+		filename = args[0]
+		if filename == "" {
+			return nil, nil, errors.New("got empty file name")
+		}
+		ipRangeStart = net.ParseIP(args[1])
+		if ipRangeStart.To4() == nil {
+			return nil, nil, errors.New("expected an IP address, got: " + args[1])
+		}
+		ipRangeEnd = net.ParseIP(args[2])
+		if ipRangeEnd.To4() == nil {
+			return nil, nil, errors.New("expected an IP address, got: " + args[2])
+		}
+		if binary.BigEndian.Uint32(ipRangeStart.To4()) >= binary.BigEndian.Uint32(ipRangeEnd.To4()) {
+			return nil, nil, errors.New("start of IP range has to be lower than the end fo an IP range")
+		}
+		var err error
+		LeaseTime, err = time.ParseDuration(args[3])
+		if err != nil {
+			return Handler6, Handler4, errors.New("expected an uint32, got: " + args[3])
+		}
```

Comment:
I would do the `if v6 {` here since this is where it starts the v4 specific part.